liblangutil/ErrorReporter.h

```diff
@@ -143,6 +143,28 @@ class ErrorReporter
 	// @returns true if the maximum error count has been reached.
 	bool hasExcessiveErrors() const;
 
+	class ErrorWatcher
+	{
+	public:
+		ErrorWatcher(ErrorReporter const& _errorReporter):
+			m_errorReporter(_errorReporter),
+			m_initialErrorCount(_errorReporter.errorCount())
+		{}
+		bool operator ()() const
```

Comment:
What do you think about `operator bool() const` instead? This would mean that we could return `watcher` instead of `watcher()`.

Reply:
He considered that, but we concluded that it should be a bit more explicit.

Reply:
@hrkrshnn 

Indeed, `operator bool()` was considered. The implementation travels between simplicity and accuracy, some variants are:

```
struct ErrorWatcher
{
    ErrorReporter const& errorReporter;
    unsigned const initialErrorCount = 0;
    bool success() const { return initialErrorCount == errorReporter.errorCount(); }
    operator bool() const { return success(); }
};
```

```
class ErrorWatcher
{
private:
    ErrorWatcher(ErrorReporter const& _errorReporter):
        m_errorReporter(_errorReporter),
        m_initialErrorCount(_errorReporter.errorCount())
    {}
public:
    bool success() const
    {
        solAssert(m_initialErrorCount <= m_errorReporter.errorCount(), "Unexpected error count.");
        return m_initialErrorCount == m_errorReporter.errorCount();
    }
    operator bool() const { return success(); }
private:
    ErrorReporter const& m_errorReporter;
    unsigned const m_initialErrorCount;

    friend ErrorReporter;
};
```